src/go/rpk/pkg/config/rpk_yaml.go

```diff
@@ -274,12 +342,42 @@ func (p *RpkProfile) HasSASLCredentials() bool {
 	return s != nil && s.User != "" && s.Password != ""
 }
 
+// ActualAuth returns the actual cloud auth for this profile, if it exists.
+func (p *RpkProfile) ActualAuth() *RpkCloudAuth {
+	if p == nil {
+		return nil
+	}
+	return p.c.rpkYamlActual.LookupAuth(p.CloudCluster.AuthOrgID, p.CloudCluster.AuthKind)
+}
+
+// VirtualAuth returns the virtual cloud auth for this profile if
+// this profile is for a cloud cluster (following the newer scheme
+// of the CloudCluster field).
+func (p *RpkProfile) VirtualAuth() *RpkCloudAuth {
+	if p == nil {
+		return nil
+	}
+	return p.c.rpkYaml.LookupAuth(p.CloudCluster.AuthOrgID, p.CloudCluster.AuthKind)
+}
+
 // HasClientCredentials returns if both ClientID and ClientSecret are empty.
 func (a *RpkCloudAuth) HasClientCredentials() bool {
 	k, _ := a.Kind()
 	return k == CloudAuthClientCredentials
 }
 
+// Equals returns if the two cloud auths are the same, which is true
+// if the name and org ID match (ignoring all other values).
+func (a *RpkCloudAuth) Equals(other *RpkCloudAuth) bool {
+	if a == nil && other == nil {
+		return true
+	}
+	if a == nil || other == nil {
+		return false
+	}
+	return a.Name == other.Name && a.OrgID == other.OrgID
```

Comment:
This should be OrgID and Kind (not name)

Reply:
Actually, changing this to Name == Name -- name has org name, ID, and kind embedded.